### PR TITLE
[PDI-18643] Cassandra Output : Time To Live(TTL) value is not capture…

### PR DESCRIPTION
**Pull request by ppatricio**
…d while loading data from Spoon.

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **98**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-cassandra-plugin/job/PR-147/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **98**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-cassandra-plugin/job/PR-147/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **98**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-cassandra-plugin/job/PR-147/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 52s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **98**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-cassandra-plugin/job/PR-147/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/trans/steps/cassandraoutput/CassandraOutput.java [hunk 1]
```diff
@@ -373,6 +373,10 @@ protected void doBatch( List<Object[]> batch ) throws Exception {
       int rowsAdded = 0;
       batch = CassandraUtils.fixBatchMismatchedTypes( batch, getInputRowMeta(), m_cassandraMeta );
       DriverCQLRowHandler handler = (DriverCQLRowHandler) m_cqlHandler;
+      String ttl = m_meta.getTTL();
+      if ( !Utils.isEmpty( ttl ) ) {
+        handler.setTtlSec( Integer.parseInt( ttl ) );
```
Comment (ssamora) [inline core/src/main/java/org/pentaho/di/trans/steps/cassandraoutput/CassandraOutput.java:378]: Is it possible for the ttl to be different from an Integer value? (by wrong user input for example). If so, what happens here  if Interger.parseInt throws an exception?

Comment (ppatricio) [reply core/src/main/java/org/pentaho/di/trans/steps/cassandraoutput/CassandraOutput.java:378]: i recall testing this at the time, let me check if this validation is already made by our interface

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **98**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-cassandra-plugin/job/PR-147/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/trans/steps/cassandraoutput/CassandraOutput.java (hunk 1 above)
Comment (ppatricio) [reply core/src/main/java/org/pentaho/di/trans/steps/cassandraoutput/CassandraOutput.java]: This is already validated during the cassandra database connection process. That's why i did not see this before providing my fix. Thank you. New commit added using the same validation logic on this one also.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 25s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **98**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-cassandra-plugin/job/PR-147/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**